--- a/ddclient	2019-01-25 09:10:15.542582719 -0500
+++ b/ddclient	2019-01-25 11:59:05.770800458 -0500
@@ -2019,7 +2019,7 @@ sub geturl {
     $request .= "Connection: close\n";
     $request .= "$headers\n";
     $request .= "Content-Length: ".length($data)."\n" if $data;
-    $request .= "\n";
+    $request .= "\n" if $data || $headers;
     $request .= $data;
 
     ## make sure newlines are <cr><lf> for some pedantic proxy servers
