diff -Nuar a/config/Makefile.in b/config/Makefile.in
--- a/config/Makefile.in	2015-12-17 17:06:59.057268124 -0500
+++ b/config/Makefile.in	2015-12-17 17:08:52.496732693 -0500
@@ -226,11 +226,11 @@
 # Install the shell profile scrips and the system startup 
 # script in the etc directory.
 etcdir = $(prefix)/etc
-noinst_DATA = openlava.sh openlava.csh \
+noinst_DATA = openlava openlava.sh openlava.csh \
 	lsf.conf lsf.cluster.${openlavacluster} lsf.shared lsf.task \
 	lsb.hosts lsb.params lsb.queues lsb.users 
 
-etc_SCRIPTS = openlava
+etc_SCRIPTS =
 all: all-am
 
 .SUFFIXES:
@@ -481,8 +481,6 @@
 # Create the working directory where the working files are
 # kept.
 install-data-local:
-	mkdir -p $(DESTDIR)$(prefix)/work/logdir
-	mkdir -p $(DESTDIR)$(prefix)/log
 
 # Tell automake to install during distcheck the
 # configuration and etc files. 
diff -Nuar a/config/lsf.conf.in b/config/lsf.conf.in
--- a/config/lsf.conf.in	2015-12-17 17:07:29.402392366 -0500
+++ b/config/lsf.conf.in	2015-12-16 14:35:56.523801309 -0500
@@ -5,15 +5,15 @@
 #
 LSF_SERVERDIR=@prefix@/sbin
 LSF_BINDIR=@prefix@/bin
-LSB_SHAREDIR=@prefix@/work
-LSF_ENVDIR=@prefix@/etc
+LSB_SHAREDIR=/var/lib/openlava
+LSF_ENVDIR=/etc/openlava
 
 # Configuration directories
-LSF_CONFDIR=@prefix@/etc
-LSB_CONFDIR=@prefix@/etc
+LSF_CONFDIR=/etc/openlava
+LSB_CONFDIR=/etc/openlava
 
 # Daemon log messages
-LSF_LOGDIR=@prefix@/log
+LSF_LOGDIR=/var/log/openlava
 LSF_LOG_MASK=LOG_WARNING
 
 # Daemons listening ports must not
