--- a/CMakeLists.txt	2026-06-08 12:25:32.827239245 -0400
+++ b/CMakeLists.txt	2026-06-08 13:26:19.614831289 -0400
@@ -47,8 +47,9 @@
     set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
 endif()
 
-set(OLLAMA_BUILD_DIR ${CMAKE_BINARY_DIR}/lib/ollama)
-set(OLLAMA_LIB_DIR "lib/ollama" CACHE STRING "Install destination for Ollama runtime payloads")
+include(GNUInstallDirs)
+set(OLLAMA_BUILD_DIR ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/ollama)
+set(OLLAMA_LIB_DIR "${CMAKE_INSTALL_FULL_LIBDIR}/ollama" CACHE STRING "Install destination for Ollama runtime payloads")
 set(OLLAMA_INSTALL_DIR ${OLLAMA_LIB_DIR}/${OLLAMA_RUNNER_DIR})
 
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY         ${OLLAMA_BUILD_DIR})
