--- a/CMakeLists.txt.old	2018-01-25 15:00:39.298047000 -0500
+++ b/CMakeLists.txt	2018-01-25 15:11:19.855338029 -0500
@@ -73,6 +73,7 @@
 # Make sure CMake uses the correct DOLFINConfig.cmake for tests and demos
 set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH} ${CMAKE_CURRENT_BINARY_DIR}/dolfin)
 
+INCLUDE(GNUInstallDirs) 
 #------------------------------------------------------------------------------
 # Configurable options for how we want to build
 
@@ -114,10 +115,10 @@
 if (APPLE)
   # The RPATH to be used when installing, but only if it's not a
   # system directory
-  SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-  LIST(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
+  SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_FULL_LIBDIR}")
+  LIST(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_FULL_LIBDIR}" isSystemDir)
   IF("${isSystemDir}" STREQUAL "-1")
-     SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+     SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_FULL_LIBDIR}")
   ENDIF("${isSystemDir}" STREQUAL "-1")
 endif()
 
@@ -796,14 +797,14 @@
 endif()
 
 # Set DOLFIN install sub-directories
-set(DOLFIN_BIN_DIR "bin" CACHE PATH "Binary installation directory.")
-set(DOLFIN_LIB_DIR "lib" CACHE PATH "Library installation directory.")
-set(DOLFIN_INCLUDE_DIR "include" CACHE PATH "C/C++ header installation directory.")
-set(DOLFIN_PKGCONFIG_DIR "lib/pkgconfig" CACHE PATH "pkg-config file installation directory.")
-set(DOLFIN_SHARE_DIR "share/dolfin" CACHE PATH "Shared data installation directory.")
-set(DOLFIN_MAN_DIR "share/man" CACHE PATH "Manual page installation directory.")
-set(DOLFIN_DOC_DIR "${DOLFIN_SHARE_DIR}/doc" CACHE PATH "DOLFIN Documentation directory.")
-set(DOLFIN_ETC_DIR "etc" CACHE PATH "Configuration file directory.")
+set(DOLFIN_BIN_DIR "${CMAKE_INSTALL_BINDIR}" CACHE PATH "Binary installation directory.")
+set(DOLFIN_LIB_DIR "${CMAKE_INSTALL_LIBDIR}" CACHE PATH "Library installation directory.")
+set(DOLFIN_INCLUDE_DIR "${CMAKE_INSTALL_INCLUDEDIR}" CACHE PATH "C/C++ header installation directory.")
+set(DOLFIN_PKGCONFIG_DIR "${CMAKE_INSTALL_LIBDIR}/pkgconfig" CACHE PATH "pkg-config file installation directory.")
+set(DOLFIN_SHARE_DIR "${CMAKE_INSTALL_DATADIR}" CACHE PATH "Shared data installation directory.")
+set(DOLFIN_MAN_DIR "${CMAKE_INSTALL_MANDIR}" CACHE PATH "Manual page installation directory.")
+set(DOLFIN_DOC_DIR "${CMAKE_INSTALL_DOCDIR}" CACHE PATH "DOLFIN Documentation directory.")
+set(DOLFIN_ETC_DIR ${CMAKE_INSTALL_SYSCONFDIR} CACHE PATH "Configuration file directory.")
 
 # Add source directory
 add_subdirectory(dolfin)
