--- /root/configure.base	2008-03-26 15:49:12.000000000 -0400
+++ src/configure.base	2008-03-26 15:55:19.000000000 -0400
@@ -383,19 +383,23 @@
 if test "$HAVE_JAVA" = "yes"; then
    AC_MSG_CHECKING([for java package in Octave])
    HAVE_OCTAVE_JAVA=[`echo "pkg('load','java');disp('yes');" | octave -qf`]
-   if test "$HAVE_OCTAVE_JAVA" = "yes"; then
-     OCTAVE_JAVA_DIR=[`echo "pkg('load','java');disp(fileparts(file_in_loadpath('javaclasspath.m')))" | octave -qf`]
-     AC_MSG_RESULT($OCTAVE_JAVA_DIR)
-     OCTAVE_JAR="${OCTAVE_JAVA_DIR}${OCTAVE_FILESEP}octave.jar"
-     OCTAVE_JAVAOCT_DIR=[`echo "pkg('load','java');disp(fileparts(file_in_loadpath('__java__.oct')))" | octave -qf`]
-     JOGL_JAR=[`echo 'jogl=file_in_path(EXEC_PATH,"jogl.jar");if (!isempty(jogl)), disp(jogl);, endif' | octave -qf`]
-     if test "x$JOGL_JAR" = "x"; then
-       JOGL_JAR=jogl.jar
-       HAVE_JAVA="Can not find jogl.jar"
-     fi
-   else
-     HAVE_JAVA="Could not find octave.jar";
-   fi
+   AC_MSG_RESULT($HAVE_OCTAVE_JAVA)
+  
+   AC_MSG_CHECKING([for Octave's java dir])
+   OCTAVE_JAVA_DIR=[`echo "pkg('load','java');disp(fileparts(file_in_loadpath('javaclasspath.m')))" | octave -qf`]
+   AC_MSG_RESULT($OCTAVE_JAVA_DIR)
+
+   AC_MSG_CHECKING([for octave.jar])
+   OCTAVE_JAR=[`java-config --with-dependencies --classpath=octave-forge-java`]
+   AC_MSG_RESULT($OCTAVE_JAR)
+
+   AC_MSG_CHECKING([for __java__.oct])
+   OCTAVE_JAVAOCT_DIR=[`echo "pkg('load','java');disp(fileparts(file_in_loadpath('__java__.oct')))" | octave -qf`]
+   AC_MSG_RESULT($OCTAVE_JAVAOCT_DIR)
+   
+   AC_MSG_CHECKING([for jogl.jar])
+   JOGL_JAR=[`java-config --with-dependencies --classpath=jogl`]
+   AC_MSG_RESULT($JOGL_JAR)
 fi
 
 CONFIGURE_OUTPUTS="Makeconf"
