diff -ur jhandles-0.3.4/PKG_ADD jhandles-0.3.4-new/PKG_ADD
--- jhandles-0.3.4/PKG_ADD	2008-08-24 12:13:48.000000000 -0400
+++ jhandles-0.3.4-new/PKG_ADD	2008-11-13 00:36:27.000000000 -0500
@@ -1,8 +1,8 @@
-if (file_in_path (EXEC_PATH, "jogl.jar"))
-  if (file_in_path (EXEC_PATH, "gluegen-rt.jar"))
-    javaaddpath (file_in_path (EXEC_PATH, "gluegen-rt.jar"));
+if (file_in_path ("/usr/share/jogl/lib", "jogl.jar"))
+  if (file_in_path ("/usr/share/gluegen/lib", "gluegen-rt.jar"))
+    javaaddpath (file_in_path ("/usr/share/gluegen/lib", "gluegen-rt.jar"));
   endif
-  javaaddpath (file_in_path (EXEC_PATH, "jogl.jar"));
+  javaaddpath (file_in_path ("/usr/share/jogl/lib", "jogl.jar"));
   javaaddpath (file_in_loadpath ("jhandles.jar"));
 
   dispatch ("get", "__jhandles_get", "any");
diff -ur jhandles-0.3.4/src/configure jhandles-0.3.4-new/src/configure
--- jhandles-0.3.4/src/configure	2008-08-24 12:37:20.000000000 -0400
+++ jhandles-0.3.4-new/src/configure	2008-11-13 21:39:36.000000000 -0500
@@ -689,7 +689,6 @@
 STRIP
 JAVAC
 JAR
-JAVA_ARCH
 JAVA_HOME
 JAVA_INCS
 JAVA_LIBS
@@ -2929,7 +2928,7 @@
 
 { echo "$as_me:$LINENO: checking for OCTAVE_VERSION in Octave" >&5
 echo $ECHO_N "checking for OCTAVE_VERSION in Octave... $ECHO_C" >&6; }
-OCTAVE_VERSION=`echo "disp(OCTAVE_VERSION)" | $OCTAVE -qf`
+OCTAVE_VERSION=`echo "disp(OCTAVE_VERSION)" | TERM="" $OCTAVE -qf`
 { echo "$as_me:$LINENO: result: $OCTAVE_VERSION" >&5
 echo "${ECHO_T}$OCTAVE_VERSION" >&6; }
 
@@ -2937,7 +2936,7 @@
 
 { echo "$as_me:$LINENO: checking for octave_config_info('canonical_host_type') in Octave" >&5
 echo $ECHO_N "checking for octave_config_info('canonical_host_type') in Octave... $ECHO_C" >&6; }
-canonical_host_type=`echo "disp(octave_config_info('canonical_host_type'))" | $OCTAVE -qf`
+canonical_host_type=`echo "disp(octave_config_info('canonical_host_type'))" | TERM="" $OCTAVE -qf`
 { echo "$as_me:$LINENO: result: $canonical_host_type" >&5
 echo "${ECHO_T}$canonical_host_type" >&6; }
 
@@ -2945,7 +2944,7 @@
 
 { echo "$as_me:$LINENO: checking for octave_config_info('SHLEXT') in Octave" >&5
 echo $ECHO_N "checking for octave_config_info('SHLEXT') in Octave... $ECHO_C" >&6; }
-SHLEXT=`echo "disp(octave_config_info('SHLEXT'))" | $OCTAVE -qf`
+SHLEXT=`echo "disp(octave_config_info('SHLEXT'))" | TERM="" $OCTAVE -qf`
 { echo "$as_me:$LINENO: result: $SHLEXT" >&5
 echo "${ECHO_T}$SHLEXT" >&6; }
 
@@ -3133,7 +3132,6 @@
 
 
 
-
 # Extract the first word of "java", so it can be a program name with args.
 set dummy java; ac_word=$2
 { echo "$as_me:$LINENO: checking for $ac_word" >&5
@@ -3267,23 +3265,15 @@
             JAVA_HOME=/usr/lib/jvm
           fi
           JAVA_HOME=`echo -n $JAVA_HOME | sed -e 's|/$||'`
-          if test -d "${JAVA_HOME}/jre/lib/i386"; then
-            JAVA_ARCH="i386";
-          elif test -d "${JAVA_HOME}/jre/lib/amd64"; then
-            JAVA_ARCH="amd64"
-          elif test -d "${JAVA_HOME}/jre/lib/mac"; then
-            JAVA_ARCH="mac"
-          elif test -d "${JAVA_HOME}/jre/lib/maci"; then
-            JAVA_ARCH="maci"
-          elif test -d "${JAVA_HOME}/jre/lib/solaris"; then
-            JAVA_ARCH="solaris"
-          elif test -d "${JAVA_HOME}/jre/lib/solarisv9"; then
+          if test -d "${JAVA_HOME}/jre/lib/solarisv9"; then
             JAVA_LIBS="solarisv9"
-          fi
-          if test -n "$JAVA_ARCH"; then
+          else if test -d "${JAVA_HOME}/include"; then
             HAVE_JAVA=yes
             JAVA_INCS="-I${JAVA_HOME}/include -I${JAVA_HOME}/include/linux"
-          fi
+	  else
+	    { echo "$as_me:$LINENO: WARNING: need JDK for /include dir" >&5
+echo "$as_me: WARNING: need JDK for /include dir" >&2;}
+          fi fi
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
@@ -4211,14 +4201,14 @@
 
 { echo "$as_me:$LINENO: checking for filesep in Octave" >&5
 echo $ECHO_N "checking for filesep in Octave... $ECHO_C" >&6; }
-OCTAVE_FILESEP=`echo "disp(filesep)" | $OCTAVE -qf`
+OCTAVE_FILESEP=`echo "disp(filesep)" | TERM="" $OCTAVE -qf`
 { echo "$as_me:$LINENO: result: $OCTAVE_FILESEP" >&5
 echo "${ECHO_T}$OCTAVE_FILESEP" >&6; }
 
 
 { echo "$as_me:$LINENO: checking for pathsep in Octave" >&5
 echo $ECHO_N "checking for pathsep in Octave... $ECHO_C" >&6; }
-OCTAVE_PATHSEP=`echo "disp(pathsep)" | $OCTAVE -qf`
+OCTAVE_PATHSEP=`echo "disp(pathsep)" | TERM="" $OCTAVE -qf`
 { echo "$as_me:$LINENO: result: $OCTAVE_PATHSEP" >&5
 echo "${ECHO_T}$OCTAVE_PATHSEP" >&6; }
 
@@ -4234,7 +4224,7 @@
 echo "${ECHO_T}$OCTAVE_JAVA_DIR" >&6; }
      OCTAVE_JAR="${OCTAVE_JAVA_DIR}${OCTAVE_FILESEP}octave.jar"
      OCTAVE_JAVAOCT_DIR=`unset TERM; echo "pkg('load','java');disp(fileparts(file_in_loadpath('__java__.oct')))" | octave -qf`
-     JOGL_JAR=`unset TERM; echo 'jogl=file_in_path(EXEC_PATH,"jogl.jar");if (!isempty(jogl)), disp(jogl);, endif' | octave -qf`
+     JOGL_JAR=`unset TERM; echo 'jogl=file_in_path("/usr/share/jogl/lib","jogl.jar");if (!isempty(jogl)), disp(jogl);, endif' | octave -qf`
      if test "x$JOGL_JAR" = "x"; then
        JOGL_JAR=jogl.jar
        HAVE_JAVA="Can not find jogl.jar"
@@ -4900,7 +4890,6 @@
 STRIP!$STRIP$ac_delim
 JAVAC!$JAVAC$ac_delim
 JAR!$JAR$ac_delim
-JAVA_ARCH!$JAVA_ARCH$ac_delim
 JAVA_HOME!$JAVA_HOME$ac_delim
 JAVA_INCS!$JAVA_INCS$ac_delim
 JAVA_LIBS!$JAVA_LIBS$ac_delim
@@ -4921,7 +4910,7 @@
 LTLIBOBJS!$LTLIBOBJS$ac_delim
 _ACEOF
 
-  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 95; then
+  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 94; then
     break
   elif $ac_last_try; then
     { { echo "$as_me:$LINENO: error: could not make $CONFIG_STATUS" >&5
diff -ur jhandles-0.3.4/src/configure.base jhandles-0.3.4-new/src/configure.base
--- jhandles-0.3.4/src/configure.base	2008-08-24 12:13:46.000000000 -0400
+++ jhandles-0.3.4-new/src/configure.base	2008-11-13 21:39:36.000000000 -0500
@@ -241,9 +241,12 @@
 dnl
 dnl OCTAVE_EVAL(expr,var) -> var=expr
 dnl
+dnl need to clear TERM variable so readline doesn't spit out a spurious <esc>1034h
+dnl http://www.nabble.com/spurious-characters-to-stdout-td15755083.html
+dnl
 AC_DEFUN(OCTAVE_EVAL,
 [AC_MSG_CHECKING([for $1 in Octave])
-$2=`echo "disp($1)" | $OCTAVE -qf`
+$2=`echo "disp($1)" | TERM="" $OCTAVE -qf`
 AC_MSG_RESULT($$2)
 AC_SUBST($2)
 ])
@@ -308,7 +311,6 @@
 
 AC_SUBST(JAVAC)
 AC_SUBST(JAR)
-AC_SUBST(JAVA_ARCH)
 AC_SUBST(JAVA_HOME)
 AC_SUBST(JAVA_INCS)
 AC_SUBST(JAVA_LIBS)
@@ -344,23 +346,14 @@
             JAVA_HOME=/usr/lib/jvm
           fi
           JAVA_HOME=[`echo -n $JAVA_HOME | sed -e 's|/$||'`]
-          if test -d "${JAVA_HOME}/jre/lib/i386"; then
-            JAVA_ARCH="i386";
-          elif test -d "${JAVA_HOME}/jre/lib/amd64"; then
-            JAVA_ARCH="amd64"
-          elif test -d "${JAVA_HOME}/jre/lib/mac"; then
-            JAVA_ARCH="mac"
-          elif test -d "${JAVA_HOME}/jre/lib/maci"; then
-            JAVA_ARCH="maci"
-          elif test -d "${JAVA_HOME}/jre/lib/solaris"; then
-            JAVA_ARCH="solaris"
-          elif test -d "${JAVA_HOME}/jre/lib/solarisv9"; then
+          if test -d "${JAVA_HOME}/jre/lib/solarisv9"; then
             JAVA_LIBS="solarisv9"
-          fi
-          if test -n "$JAVA_ARCH"; then
+          else if test -d "${JAVA_HOME}/include"; then
             HAVE_JAVA=yes
             JAVA_INCS="-I${JAVA_HOME}/include -I${JAVA_HOME}/include/linux"
-          fi
+	  else
+	    AC_MSG_WARN([need JDK for /include dir])
+          fi fi
           AC_CHECK_HEADER(GL/gl.h, [], AC_MSG_ERROR(Could not find OpenGL headers))
           OF_CHECK_LIB(GL, glEnable, OPENGL_LIBS=-lGL, [
 	    ## Invalidate the cache.
@@ -388,7 +381,7 @@
      AC_MSG_RESULT($OCTAVE_JAVA_DIR)
      OCTAVE_JAR="${OCTAVE_JAVA_DIR}${OCTAVE_FILESEP}octave.jar"
      OCTAVE_JAVAOCT_DIR=[`unset TERM; echo "pkg('load','java');disp(fileparts(file_in_loadpath('__java__.oct')))" | octave -qf`]
-     JOGL_JAR=[`unset TERM; echo 'jogl=file_in_path(EXEC_PATH,"jogl.jar");if (!isempty(jogl)), disp(jogl);, endif' | octave -qf`]
+     JOGL_JAR=[`unset TERM; echo 'jogl=file_in_path("/usr/share/jogl/lib","jogl.jar");if (!isempty(jogl)), disp(jogl);, endif' | octave -qf`]
      if test "x$JOGL_JAR" = "x"; then
        JOGL_JAR=jogl.jar
        HAVE_JAVA="Can not find jogl.jar"
